pkg/storage/s3.go

```diff
@@ -404,61 +409,129 @@ func (rs *S3Storage) WalkDir(ctx context.Context, opt *WalkOption, fn func(strin
 
 // Open a Reader by file path.
 func (rs *S3Storage) Open(ctx context.Context, path string) (ReadSeekCloser, error) {
-	reader, err := rs.open(ctx, path, 0, 0)
+	reader, r, err := rs.open(ctx, path, 0, 0)
 	if err != nil {
 		return nil, err
 	}
 	return &s3ObjectReader{
-		storage: rs,
-		name:    path,
-		reader:  reader,
+		storage:   rs,
+		name:      path,
+		reader:    reader,
+		ctx:       ctx,
```

Comment:
Urgh, you're voiding the progress of #452. At this point it's better to change the method to be `SeekContext(ctx context.Context, offset int64, whence int) error` and abandon the Seeker interface.

Reply:
this ctx is mainly using for read parquet files, because parquet reader need open multi reader for each column, if we change this interface, how can we met the parquet file read requirement here: https://github.com/xitongsys/parquet-go/blob/master/source/source.go#L9-L16